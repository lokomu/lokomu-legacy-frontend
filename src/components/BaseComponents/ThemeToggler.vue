<template>
  <div>
    <MoonIcon
      class="fill-floyd-800 h-7 cursor-pointer dark:hidden"
      @click="toggleTheme"
    />
    <SunIcon
      class="fill-yellow-300 h-7 cursor-pointer hidden dark:block"
      @click="toggleTheme"
    />
  </div>
</template>

<script setup lang="ts">
import { useThemeStore } from "@/stores/ThemeStore";
import MoonIcon from "@/components/IconComponents/MoonIcon.vue";
import SunIcon from "@/components/IconComponents/SunIcon.vue";

const themeStore = useThemeStore();

function toggleTheme(): void {
  themeStore.toggleTheme();
}
</script>
