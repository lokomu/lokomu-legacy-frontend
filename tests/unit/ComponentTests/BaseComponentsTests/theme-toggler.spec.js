import { shallowMount } from "@vue/test-utils";
import ThemeToggler from "@/components/BaseComponents/ThemeToggler";
import { describe, expect, it, beforeEach } from "vitest";

describe("ThemeToggler component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(ThemeToggler);
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("is instantiated", () => {
    expect(wrapper.exists()).toBeTruthy();
  });
});
