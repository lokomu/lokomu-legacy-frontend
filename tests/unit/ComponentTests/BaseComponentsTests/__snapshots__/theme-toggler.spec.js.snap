// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ThemeToggler component > renders correctly 1`] = `
<div>
  <moon-icon-stub
    class="fill-floyd-800 h-7 cursor-pointer dark:hidden"
  />
  <sun-icon-stub
    class="fill-yellow-300 h-7 cursor-pointer hidden dark:block"
  />
</div>
`;
